packages/react/src/components/AssessShared/EnterEmailModal.tsx

```diff
@@ -0,0 +1,92 @@
+import { Box, Button, FormField, Input, Modal, SpaceBetween } from "@awsui/components-react";
+import { MemoryHistory } from "history";
+import React from "react";
+import { useHistory } from "react-router-dom";
+
+interface Props {
+  visible: boolean;
+}
+
+// This function simply checks if the email is currently set
+// It is designed to be used when implementing the EnterEmailModal
+// However, it does not have to be used if the implementer
+// wants to use their own functionality
+// Use it to determine whether or not to display the modal
+export const isEmailSet = () => {
+  const email = window.electron.getState("email");
+  if (email === "" || email == null) {
+    return false;
+  }
+  return true;
+};
+
+export const EnterEmailModal: React.FC<Props> = React.memo(({ visible }) => {
+  const [isVisible, setVisible] = React.useState(visible);
+  const [emailValue, setEmailValue] = React.useState("");
+  const [emailError, setEmailError] = React.useState("");
+  const history = useHistory() as MemoryHistory;
+
+  // Used to check if the email string is valid, e.g. example@amazon.com
+  const validEmail = new RegExp(`^[^\\s@]+@[^\\s@]+\\.[^\\s@]+$`);
+
+  const saveAndClose = () => {
+    // Validate
+    if (validEmail.test(emailValue)) {
+      // email is valid, save
+      window.electron.saveState("email", emailValue);
+      setVisible(false);
+    } else {
+      // invalid email
+      setEmailError("Invalid e-mail format.");
+    }
+  };
+
+  const exitWithoutSave = () => {
+    setVisible(false);
+    history.go(-1);
+  };
+
+  return (
+    <Modal
+      onDismiss={() => exitWithoutSave()}
+      visible={isVisible}
+      closeAriaLabel="Close modal"
+      size="medium"
+      footer={
+        <Box float="right">
+          <SpaceBetween direction="horizontal" size="xs">
+            <Button variant="link" onClick={() => exitWithoutSave()}>
+              Cancel
+            </Button>
+            <Button variant="primary" onClick={() => saveAndClose()}>
+              Save
+            </Button>
+          </SpaceBetween>
+        </Box>
+      }
+      header="E-mail required"
+    >
+      <SpaceBetween direction="vertical" size="m">
+        <Box>
+          We require your e-mail to use this feature. It looks like you have not provided your email yet. Please enter
```

Comment:
Has this text been approved by the UI team?

Reply:
Approved by legal; will check with UX

Reply:
Approved by UX